File: SampleProject/SampleProject/UIViewController+AlertHelper.m
//
//  UIViewController+AlertHelper.m
//  SampleProject
//
//  Created by Fotis Mitropoulos on 25/8/20.
//  Copyright © 2020 POLLFISH. All rights reserved.
//

#import "UIViewController+AlertHelper.h"

@implementation UIViewController (AlertExtension)

- (void)showNoPermissionAlert {
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Pollfish"
                                                                             message:@"Unfortunately Pollfish cannot proceed without the Tracking Permission. Please go to your Application Settings and enable \"Allow Tracking\"."
                                                                      preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *actionClose = [UIAlertAction actionWithTitle:@"Close"
                                                          style:UIAlertActionStyleDefault
                                                        handler:nil];
    
    UIAlertAction *actionGoToSettings = [UIAlertAction actionWithTitle:@"Go to Settings"
                                                                 style:UIAlertActionStyleDefault
                                                               handler:^(UIAlertAction* action) {
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString]
                                           options:@{}
                                 completionHandler:nil];
    }];
    
    [alertController addAction:actionClose];
    [alertController addAction:actionGoToSettings];
    [self presentViewController:alertController animated:YES completion:nil];
}

@end

File: SampleProjectSwift/SampleProjectSwift/UIViewController+AlertHelper.swift
//
//  UIViewController+PermissionHelper.swift
//  SampleProjectSwift
//
//  Created by Fotis Mitropoulos on 25/8/20.
//  Copyright © 2020 POLLFISH. All rights reserved.
//

import UIKit

extension UIViewController {
    
    func showNoPermissionAlert() {
        let alertController = UIAlertController(title: "Pollfish",
                                                message: "Unfortunately Pollfish cannot proceed without the Tracking Permission. Please go to your Application Settings and enable \"Allow Tracking\".",
                                                preferredStyle: .alert)
        
        let actionClose = UIAlertAction(title: "Close",
                                        style: .default)
        
        let actionGoToSettings = UIAlertAction(title: "Go to Settings", style: .default) { action in
            if #available(iOS 10, *) {
                UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!, options: [UIApplication.OpenExternalURLOptionsKey:Any]())
            } else {
                UIApplication.shared.openURL(URL(string: UIApplication.openSettingsURLString)!)
            }
        }
        
        alertController.addAction(actionClose)
        alertController.addAction(actionGoToSettings)
        
        present(alertController, animated: true)
    }
    
}
